### PR TITLE
refactor: migrate from MDX to CSF stories with autodocs

### PR DESCRIPTION
**Pull request by rsmelo92**
## Done

This PR migrates Card component stories from MDX to CSF, if accepted I will do it with all remaining components

- Created a CSF file for rendering stories of the card component
- Allowed [autodocs](https://storybook.js.org/docs/writing-docs/autodocs) on it
- Deleted MDX version of Card's stories

## Screenshots

https://github.com/canonical/react-components/assets/16295402/9cb0dd0f-146d-47bf-b6c1-371fc0d37818

### QA steps

- Run the project
- Open Card stories
- Check that the behaviors haven't changed
- Open other stories
- Check that they didn't break as well
- Open the code
- Check that there is less code, less cognitive overload and it has improved

## Fixes

Fixes: #1001  .

### CODE REVIEW

File: .storybook/preview.js
Comment (rsmelo92) [reply .storybook/preview.js:13]: the custom page is here to keep the same layout that is being used today, otherwise the layout adds a `<Primary />` example, and instead of `<Argtypes />` it uses `<Controls />`

### Without custom page

image


### With custom page

image

File: src/components/Card/Card.stories.tsx
Comment (rsmelo92) [reply src/components/Card/Card.stories.tsx]: Thanks for the comments!

Yeah, it makes sense! Changed it in [85d9a3b](https://github.com/canonical/react-components/pull/1010/commits/85d9a3bfc5ab13a2134868a44a55601689100e2e)

File: .storybook/preview.js
Comment (bartaz) [reply .storybook/preview.js:13]: I don't think we are strongly attached to current page layout. If CSF defaults are different (with primary example on top) I think we better stick to defaults first, and then see if we want to change anything.

**Comment by bartaz** [comment]
Thanks for the PR, I kicked off CI and demo for it: https://react-components-1010.demos.haus/?path=/docs/card--docs

File: .storybook/preview.js
Comment (rsmelo92) [reply .storybook/preview.js]: agree, I believe the objective here is to lower cognitive overload when giving maintenance

changed to default layout on [1820f44](https://github.com/canonical/react-components/pull/1010/commits/1820f44963ec35b5032e970bf2190b2b4cbfcf8d)

**Comment by rsmelo92** [comment]
@bartaz @petermakowski I'm waiting for this PR to be merged to migrate all other components, or should I add them to this PR?

File: src/components/Card/Card.stories.tsx
Comment (bartaz) [inline src/components/Card/Card.stories.tsx:9]: How is this transformed into the description in the story? It doesn't seem to be treated as markdown? Or can it?

Right now you seem to be inserting double <br> to make it look like 2 paragraphs, while it's all inserted into one. Which is not ideal.

Comment (petermakowski) [reply src/components/Card/Card.stories.tsx:9]: I'd recommend to move this to the component file itself, storybook will pick it up and display in using autogenerated docs.

Comment (rsmelo92) [reply src/components/Card/Card.stories.tsx]: makes sense! moved to component, and removed the `<br/>` tags on [9179238](https://github.com/canonical/react-components/pull/1010/commits/9179238a515b9c84dc247fe6751b58a5984486b7)

**Comment by huwshimi** [comment]
Is there anything I can do to help this PR along? I'd like to migrate a bunch of components to this repo, but hoping this lands before I add the docs.

**Comment by rsmelo92** [comment]
@huwshimi from my side I think it's all ok, just waiting for the PR to be accepted (or not) 🙃

**Comment by huwshimi** [comment]
@bartaz @petermakowski do either of you want to do a final review or would you be happy for me to do it?

**Comment by petermakowski** [comment]
> @bartaz @petermakowski do either of you want to do a final review or would you be happy for me to do it?

I'm happy for you to do it!

**Review by bartaz — APPROVED** [review]
Sorry that we kept it stale for a while. Looks good and clean, thanks a lot!

**Comment by huwshimi** [comment]
Thanks everyone!

**Pull request merged**
